cpp/src/arrow/filesystem/s3fs.h

```diff
@@ -92,6 +92,27 @@ class S3RetryStrategy {
                                                 int64_t attempted_retries) = 0;
 };
 
+/// Wrapper for stock AWS retry strategies
+class AwsRetryStrategy : S3RetryStrategy {
```

Comment:
Does this need to be in the header? For one, we don't want to expose `Aws::` in our public API